### DISCUSSION TITLE
Not reading Physical Entities (Edges)

### DISCUSSION THREAD

**Discussion by tctheodosiou** [idaholab/moose#26867] [Q&A Meshing] [answered]
Hi,
I have created a 3D model in gmsh, and I am  trying to import it with FileMeshGenerator.
The file contains named faces and named edges (as Physical Entities). The faces are read without problem. The edges seem to be ignored.
I have read a similar discussion in Google Groups (which had an error in the gmsh file). So tried to compile the geo file from the command line (gmsh -3 infile.geo); no errors are shown and the exported msh file contains the named edges.

I attach my input files. The entity "bottom_edge" does not show in the moose model. 
When I run _mysolver-opt -i Q_model.i --mesh-only_ it doesnt seem to recognise the edge (_Mesh Edgesets: None_).

Any advice is much appriciated. Thanks!
(Attachments in zip, because the forum does not allow .i or .geo files)

**Comment by tctheodosiou** [comment]
It seems that when I apply the RenameBlockGenerator, it loses the defined face labels (edge-names are lost in any case).
Any way to bybass this error?

` [part2]
        type = FileMeshGenerator
        file = 'part2.msh'
    []
    [part2_subdomain]
        type = SubdomainIDGenerator
        input = part2
        subdomain_id = 2
    []`

The reason for needing the subdomains is that parts 1 & 2 have different elements, and I get an error about Exodus cant have different element type in same block id (something like that). By renaming the subdomains, I bypass Exodus error, but I lose the names of the faces (and therefore I cant apply BCs).

**Comment by GiudGiud** [comment, marked as answer]
Hello

We keep track of sidesets and nodesets in MOOSe.
In 3D a sideset can only be 2D. So your edges have to be stored as nodesets instead, sets of nodes. 

Guillaume
